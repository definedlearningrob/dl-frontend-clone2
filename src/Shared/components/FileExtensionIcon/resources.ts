export const imageExtensions = ['jpg', 'jpeg', 'png', 'gif'];
export const videoExtensions = ['wma', 'mpg', 'flv', 'avi', 'mp4', 'webm'];
export const documentExtensions = ['pdf', 'doc', 'docx', 'txt', 'csv', 'rtf', 'html'];
export const presentationExtensions = ['ppt', 'pptx'];
export const sheetExtensions = ['xls', 'xlsx'];
