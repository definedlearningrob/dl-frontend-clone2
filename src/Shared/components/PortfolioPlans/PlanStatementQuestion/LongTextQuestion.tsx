import SharedFormTextarea from '@shared/components/FormTextarea/FormTextarea';

type Props = {
  disabled: boolean;
  name: string;
};

export const LongTextQuestion = ({ disabled, name }: Props) => (
  <SharedFormTextarea
    disabled={disabled}
    errorMessage={undefined}
    name={name}
    textareaClassName='text-xxs xxxl:text-xs !p-xs !mb-0 disabled:bg-neutral-200'
  />
);
