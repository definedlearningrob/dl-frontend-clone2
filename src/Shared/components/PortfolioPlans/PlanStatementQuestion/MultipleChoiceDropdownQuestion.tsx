import { useField } from 'formik';
import { MultiValue } from 'react-select';

import { Select } from '@shared/components/Select';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';

type Props = {
  options: { option: string; id: string }[];
  disabled: boolean;
  name: string;
};

export const MultipleChoiceDropdownQuestion = ({ options, disabled, name }: Props) => {
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });

  const [field, _, helpers] = useField(name);

  const selectOptions = options.map((option) => ({ label: option.option, value: option.option }));

  const handleChange = (newValue: MultiValue<{ label: string; value: string | number }>) => {
    const newValues = newValue.map(({ label }) => label);
    helpers.setValue(newValues);
  };

  return (
    <Select
      defaultValue={selectOptions.filter((option) => field.value.includes(option.value))}
      isDisabled={disabled}
      isMulti={true}
      menuPortalTarget={document.body}
      options={selectOptions}
      size={isFullHD ? 'md' : 'sm'}
      wrapperClassName='flex-1'
      onChange={handleChange}
    />
  );
};
