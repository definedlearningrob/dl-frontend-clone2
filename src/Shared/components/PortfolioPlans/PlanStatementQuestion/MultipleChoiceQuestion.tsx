import { useField } from 'formik';
import { xor } from 'lodash-es';
import { ChangeEvent } from 'react';

import SharedCheckbox from '@shared/components/Checkbox/Checkbox';

type Props = {
  options: { option: string; id: string }[];
  disabled: boolean;
  name: string;
};

export const MultipleChoiceQuestion = ({ options, disabled, name }: Props) => {
  const [field, , helpers] = useField(name);

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    // adds a value to the array when it is not there or remove it when it is present
    const newValue = xor(field.value, [event.target.value]);
    helpers.setValue(newValue);
  };

  return (
    <fieldset className='p-0'>
      {options.map((option) => (
        <SharedCheckbox
          key={option.id}
          checked={field.value.includes(option.option)}
          className='mb-xs last:mb-0 !text-xs xxxl:!text-sm'
          disabled={disabled}
          id={option.id}
          label={option.option}
          value={option.option}
          onChange={handleChange}
        />
      ))}
    </fieldset>
  );
};
