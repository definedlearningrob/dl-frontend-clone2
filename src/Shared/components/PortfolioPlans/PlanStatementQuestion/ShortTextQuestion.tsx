import SharedFormTextInput from '@shared/components/FormTextInput/FormTextInput';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';

type Props = {
  disabled: boolean;
  name: string;
};

export const ShortTextQuestion = ({ disabled, name }: Props) => {
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });

  return (
    <SharedFormTextInput
      disabled={disabled}
      errorMessage={undefined}
      name={name}
      size={isFullHD ? 'md' : 'sm'}
    />
  );
};
