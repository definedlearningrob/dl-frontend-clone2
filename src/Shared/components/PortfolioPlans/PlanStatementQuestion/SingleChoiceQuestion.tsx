import { FormRadio } from '@shared/components/FormRadio';

type Props = {
  options: { option: string; id: string }[];
  disabled: boolean;
  name: string;
};

export const SingleChoiceQuestion = ({ options, disabled, name }: Props) => (
  <fieldset className='p-0' role='radiogroup'>
    {options.map((option) => (
      <FormRadio
        key={option.id}
        className='mb-xs last:mb-0 text-xs xxxl:text-sm'
        disabled={disabled}
        name={name}
        value={option.option}>
        {option.option}
      </FormRadio>
    ))}
  </fieldset>
);
