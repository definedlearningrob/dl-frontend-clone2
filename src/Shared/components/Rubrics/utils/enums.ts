export enum RUBRIC_TYPE {
  VIEWER = 'VIEWER',
  GRADER = 'GRADER',
  PREVIEW = 'PREVIEW',
}
