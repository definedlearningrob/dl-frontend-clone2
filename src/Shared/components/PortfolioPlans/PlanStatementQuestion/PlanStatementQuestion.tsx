import { useToggle } from 'react-use';
import { isEmpty } from 'lodash-es';
import { useTranslation } from 'react-i18next';
import * as Yup from 'yup';
import { Form, Formik } from 'formik';

import { STATEMENT_QUESTION_TYPE } from '@shared/resources/enums';
import { cx } from '@shared/utils/cx';
import Button from '@shared/components/Button/Button';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';

import { ShortTextQuestion } from './ShortTextQuestion';
import { LongTextQuestion } from './LongTextQuestion';
import { SingleChoiceQuestion } from './SingleChoiceQuestion';
import { MultipleChoiceQuestion } from './MultipleChoiceQuestion';
import { MultipleChoiceDropdownQuestion } from './MultipleChoiceDropdownQuestion';

type FormValues = {
  answer: string[];
};

type Props = {
  questionType: STATEMENT_QUESTION_TYPE;
  answer?: string[];
  options?: { option: string; id: string }[];
  text: string;
  onSubmit: (answer: string[]) => void;
};

const answerComponentsMap = {
  [STATEMENT_QUESTION_TYPE.SHORT_TEXT]: ShortTextQuestion,
  [STATEMENT_QUESTION_TYPE.LONG_TEXT]: LongTextQuestion,
  [STATEMENT_QUESTION_TYPE.SINGLE_CHOICE]: SingleChoiceQuestion,
  [STATEMENT_QUESTION_TYPE.MULTIPLE_CHOICE]: MultipleChoiceQuestion,
  [STATEMENT_QUESTION_TYPE.MULTIPLE_CHOICE_DROPDOWN]: MultipleChoiceDropdownQuestion,
};

export const PlanStatementQuestion = ({
  questionType,
  answer = [],
  options = [],
  text,
  onSubmit,
}: Props) => {
  const { t } = useTranslation();
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });

  const [isEditing, toggleIsEditing] = useToggle(isEmpty(answer));

  const AnswerComponent = answerComponentsMap[questionType];

  const handleSave = async (values: FormValues) => {
    await onSubmit(values.answer);
    toggleIsEditing();
  };

  const validationSchema = Yup.object().shape({
    answer: Yup.mixed().when({
      is: Array.isArray,
      then: () => Yup.array().min(1, t('validation.messages.required')),
      otherwise: () => Yup.string().required(t('validation.messages.required')),
    }),
  });

  const buttonSize = isFullHD ? 'md' : 'sm';

  const headingClasses = cx(
    'p-xs pt-x xxxl:p-sm xxxl:pb-x bg-neutral-200 rounded-t-xs',
    'text-xs xxxl:text-sm font-medium leading-lg'
  );

  const isDropdownQuestion = questionType === STATEMENT_QUESTION_TYPE.MULTIPLE_CHOICE_DROPDOWN;
  const cardContentClasses = cx('flex py-sm px-x xxxl:px-sm', {
    'items-center gap-sm': isDropdownQuestion,
    'flex-col gap-xs': !isDropdownQuestion,
  });

  return (
    <Formik
      initialValues={{ answer }}
      validateOnChange={true}
      validateOnMount={true}
      validationSchema={validationSchema}
      onSubmit={handleSave}>
      {({ isValid }) => (
        <Form>
          <div className='border border-neutral-300 rounded-xs'>
            <div className={headingClasses}>{text}</div>
            <div className={cardContentClasses}>
              <AnswerComponent disabled={!isEditing} name='answer' options={options} />
              <div className='flex justify-end'>
                {!isEditing && (
                  <Button
                    className='w-[80px]'
                    size={buttonSize}
                    variant='primary-outlined'
                    onClick={toggleIsEditing}>
                    {t('common.actions.edit')}
                  </Button>
                )}
                {isEditing && (
                  <Button
                    className='w-[80px]'
                    disabled={!isValid}
                    size={buttonSize}
                    type='submit'
                    variant='primary'>
                    {t('common.actions.save')}
                  </Button>
                )}
              </div>
            </div>
          </div>
        </Form>
      )}
    </Formik>
  );
};
