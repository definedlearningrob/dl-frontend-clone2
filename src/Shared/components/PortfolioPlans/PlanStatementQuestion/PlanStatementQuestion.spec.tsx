import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { renderWithI18N } from '@dc/utils/test';

import { STATEMENT_QUESTION_TYPE } from '@shared/resources/enums';

import { PlanStatementQuestion } from './PlanStatementQuestion';

const defaultOptions = [
  {
    option: 'First option',
    id: '1',
  },
  {
    option: 'Second option',
    id: '2',
  },
  { option: 'Third option', id: '3' },
];

const onSubmit = jest.fn();

describe('StatementQuestion', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  describe('Short text', () => {
    const questionType = STATEMENT_QUESTION_TYPE.SHORT_TEXT;

    it('displays text input without initial answer', async () => {
      const { container } = renderWithI18N(
        <PlanStatementQuestion
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const textInput = screen.getByRole('textbox');
      expect(textInput).toHaveValue('');

      const saveButton = screen.getByRole('button', { name: 'Save' });
      await waitFor(() => {
        expect(saveButton).toBeDisabled();
      });

      expect(container).toMatchSnapshot();
    });

    it('displays text input with initial answer', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['Initial answer']}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const textInput = screen.getByRole('textbox');
      expect(textInput).toHaveValue('Initial answer');
      expect(textInput).toBeDisabled();
    });

    it('enables edit mode when clicking "Edit" button', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['Initial answer']}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const editButton = screen.getByRole('button', { name: 'Edit' });
      userEvent.click(editButton);

      const textInput = screen.getByRole('textbox');
      expect(textInput).not.toBeDisabled();
    });

    it('fires save action with correct values', async () => {
      renderWithI18N(
        <PlanStatementQuestion
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const textInput = screen.getByRole('textbox');
      userEvent.paste(textInput, 'Example answer');
      userEvent.click(screen.getByRole('button', { name: 'Save' }));

      await waitFor(() => {
        expect(onSubmit).toHaveBeenCalledTimes(1);
      });
      expect(onSubmit).toHaveBeenCalledWith('Example answer');
      expect(screen.getByRole('button', { name: 'Edit' })).toBeInTheDocument();
      expect(textInput).toBeDisabled();
    });
  });

  describe('Long text', () => {
    const questionType = STATEMENT_QUESTION_TYPE.LONG_TEXT;

    it('displays text area without initial answer', async () => {
      const { container } = renderWithI18N(
        <PlanStatementQuestion
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const textarea = screen.getByRole('textbox');
      expect(textarea).toHaveValue('');

      const saveButton = screen.getByRole('button', { name: 'Save' });
      await waitFor(() => {
        expect(saveButton).toBeDisabled();
      });

      expect(container).toMatchSnapshot();
    });

    it('displays text area with initial answer', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['Initial answer']}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const textarea = screen.getByRole('textbox');
      expect(textarea).toHaveValue('Initial answer');
      expect(textarea).toBeDisabled();
    });

    it('enables edit mode when clicking "Edit" button', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['Initial answer']}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const editButton = screen.getByRole('button', { name: 'Edit' });
      userEvent.click(editButton);

      const textarea = screen.getByRole('textbox');
      expect(textarea).not.toBeDisabled();
    });

    it('fires save action with correct values', async () => {
      renderWithI18N(
        <PlanStatementQuestion
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const textarea = screen.getByRole('textbox');
      userEvent.paste(textarea, 'Example answer');
      userEvent.click(screen.getByRole('button', { name: 'Save' }));

      await waitFor(() => {
        expect(onSubmit).toHaveBeenCalledTimes(1);
      });
      expect(onSubmit).toHaveBeenCalledWith('Example answer');
      expect(screen.getByRole('button', { name: 'Edit' })).toBeInTheDocument();
      expect(textarea).toBeDisabled();
    });
  });

  describe('Single choice', () => {
    const questionType = STATEMENT_QUESTION_TYPE.SINGLE_CHOICE;

    it('displays radio buttons without initial answer', async () => {
      const { container } = renderWithI18N(
        <PlanStatementQuestion
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const radioButtons = screen.getAllByRole('radio');
      expect(radioButtons).toHaveLength(3);

      expect(radioButtons[0]).toHaveAccessibleName('First option');
      expect(radioButtons[1]).toHaveAccessibleName('Second option');
      expect(radioButtons[2]).toHaveAccessibleName('Third option');

      const saveButton = screen.getByRole('button', { name: 'Save' });
      await waitFor(() => {
        expect(saveButton).toBeDisabled();
      });

      expect(container).toMatchSnapshot();
    });

    it('displays radio buttons with initial answer', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['Second option']}
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const radioButtons = screen.getAllByRole('radio');
      expect(radioButtons).toHaveLength(3);

      expect(radioButtons[0]).not.toBeChecked();
      expect(radioButtons[1]).toBeChecked();
      expect(radioButtons[2]).not.toBeChecked();

      expect(radioButtons[0]).toBeDisabled();
      expect(radioButtons[1]).toBeDisabled();
      expect(radioButtons[2]).toBeDisabled();
    });

    it('enables edit mode when clicking "Edit" button', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['2']}
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const editButton = screen.getByRole('button', { name: 'Edit' });
      userEvent.click(editButton);

      const firstRadioButton = screen.getByRole('radio', { name: 'First option' });
      expect(firstRadioButton).not.toBeDisabled();
    });

    it('fires save action with correct values', async () => {
      renderWithI18N(
        <PlanStatementQuestion
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const thirdRadioButton = screen.getByRole('radio', { name: 'Third option' });
      userEvent.click(thirdRadioButton);
      userEvent.click(screen.getByRole('button', { name: 'Save' }));

      await waitFor(() => {
        expect(onSubmit).toHaveBeenCalledTimes(1);
      });
      expect(onSubmit).toHaveBeenCalledWith(['Third option']);
      expect(screen.getByRole('button', { name: 'Edit' })).toBeInTheDocument();
      expect(thirdRadioButton).toBeDisabled();
    });
  });

  describe('Multiple choice', () => {
    const questionType = STATEMENT_QUESTION_TYPE.MULTIPLE_CHOICE;

    it('displays checkboxes without initial answer', async () => {
      const { container } = renderWithI18N(
        <PlanStatementQuestion
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const checkboxes = screen.getAllByRole('checkbox');
      expect(checkboxes).toHaveLength(3);

      expect(checkboxes[0]).toHaveAccessibleName('First option');
      expect(checkboxes[1]).toHaveAccessibleName('Second option');
      expect(checkboxes[2]).toHaveAccessibleName('Third option');

      const saveButton = screen.getByRole('button', { name: 'Save' });
      await waitFor(() => {
        expect(saveButton).toBeDisabled();
      });

      expect(container).toMatchSnapshot();
    });

    it('displays checkboxes with initial answer', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['First option', 'Second option']}
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const checkboxes = screen.getAllByRole('checkbox');
      expect(checkboxes).toHaveLength(3);

      expect(checkboxes[0]).toBeChecked();
      expect(checkboxes[1]).toBeChecked();
      expect(checkboxes[2]).not.toBeChecked();

      expect(checkboxes[0]).toBeDisabled();
      expect(checkboxes[1]).toBeDisabled();
      expect(checkboxes[2]).toBeDisabled();
    });

    it('enables edit mode when clicking "Edit" button', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['First option', 'Second option']}
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const editButton = screen.getByRole('button', { name: 'Edit' });
      userEvent.click(editButton);

      const firstCheckbox = screen.getByRole('checkbox', { name: 'First option' });
      expect(firstCheckbox).not.toBeDisabled();
    });

    it('fires save action with correct values', async () => {
      renderWithI18N(
        <PlanStatementQuestion
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const [firstCheckbox, , thirdCheckbox] = screen.getAllByRole('checkbox');
      userEvent.click(firstCheckbox);
      userEvent.click(thirdCheckbox);
      userEvent.click(screen.getByRole('button', { name: 'Save' }));

      await waitFor(() => {
        expect(onSubmit).toHaveBeenCalledTimes(1);
      });
      expect(onSubmit).toHaveBeenCalledWith(['First option', 'Third option']);
      expect(screen.getByRole('button', { name: 'Edit' })).toBeInTheDocument();
      expect(firstCheckbox).toBeDisabled();
    });
  });

  describe('Multiple choice dropdown', () => {
    const questionType = STATEMENT_QUESTION_TYPE.MULTIPLE_CHOICE_DROPDOWN;

    it('displays select without initial answer', async () => {
      const { container } = renderWithI18N(
        <PlanStatementQuestion
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const select = screen.getByRole('combobox');
      expect(select).toHaveValue('');

      const saveButton = screen.getByRole('button', { name: 'Save' });
      await waitFor(() => {
        expect(saveButton).toBeDisabled();
      });

      expect(container).toMatchSnapshot();
    });

    it('displays select with initial answer', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['First option', 'Second option']}
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const select = screen.queryByRole('combobox');
      // select input is hidden when it is disabled
      expect(select).not.toBeInTheDocument();

      const selectedValueChip = screen.getByTestId('select-chip');
      expect(selectedValueChip).toHaveTextContent('2 selected');
    });

    it('enables edit mode when clicking "Edit" button', () => {
      renderWithI18N(
        <PlanStatementQuestion
          answer={['First option', 'Second option']}
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const editButton = screen.getByRole('button', { name: 'Edit' });
      userEvent.click(editButton);

      const select = screen.getByRole('combobox');
      expect(select).not.toBeDisabled();
    });

    it('fires save action with correct values', async () => {
      renderWithI18N(
        <PlanStatementQuestion
          options={defaultOptions}
          questionType={questionType}
          text='Example question'
          onSubmit={onSubmit}
        />
      );

      const select = screen.getByRole('combobox');
      userEvent.type(select, '{enter}');
      expect(screen.getByTestId('select-chip')).toHaveTextContent('First option');

      userEvent.type(select, '{arrowdown}{enter}');
      expect(screen.getByTestId('select-chip')).toHaveTextContent('2 selected');

      userEvent.click(screen.getByRole('button', { name: 'Save' }));

      await waitFor(() => {
        expect(onSubmit).toHaveBeenCalledTimes(1);
      });
      expect(onSubmit).toHaveBeenCalledWith(['First option', 'Second option']);
      expect(screen.getByRole('button', { name: 'Edit' })).toBeInTheDocument();
    });
  });
});
