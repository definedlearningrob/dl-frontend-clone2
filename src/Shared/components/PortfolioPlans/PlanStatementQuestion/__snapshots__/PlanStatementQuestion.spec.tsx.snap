// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatementQuestion Long text displays text area without initial answer 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="border border-neutral-300 rounded-xs"
    >
      <div
        class="p-xs pt-x xxxl:p-sm xxxl:pb-x bg-neutral-200 rounded-t-xs text-xs xxxl:text-sm font-medium leading-lg"
      >
        Example question
      </div>
      <div
        class="flex py-sm px-x xxxl:px-sm flex-col gap-xs"
      >
        <div
          class="flex flex-col gap-xs"
        >
          <label
            class="flex flex-col gap-xs"
          >
            <textarea
              class="textarea__input !mb-0 text-xxs xxxl:text-xs !p-xs !mb-0 disabled:bg-neutral-200"
              name="answer"
              placeholder="Type here..."
            />
          </label>
        </div>
        <div
          class="flex justify-end"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] w-[80px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
            data-testid="button"
            disabled=""
            type="submit"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Save
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`StatementQuestion Multiple choice displays checkboxes without initial answer 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="border border-neutral-300 rounded-xs"
    >
      <div
        class="p-xs pt-x xxxl:p-sm xxxl:pb-x bg-neutral-200 rounded-t-xs text-xs xxxl:text-sm font-medium leading-lg"
      >
        Example question
      </div>
      <div
        class="flex py-sm px-x xxxl:px-sm flex-col gap-xs"
      >
        <fieldset
          class="p-0"
        >
          <div
            class="checkbox mb-xs last:mb-0 !text-xs xxxl:!text-sm"
          >
            <input
              class="checkbox__input"
              data-testid="checkbox"
              id="1"
              tabindex="0"
              type="checkbox"
              value="First option"
            />
            <label
              class="checkbox__label"
              for="1"
            >
              First option
            </label>
          </div>
          <div
            class="checkbox mb-xs last:mb-0 !text-xs xxxl:!text-sm"
          >
            <input
              class="checkbox__input"
              data-testid="checkbox"
              id="2"
              tabindex="0"
              type="checkbox"
              value="Second option"
            />
            <label
              class="checkbox__label"
              for="2"
            >
              Second option
            </label>
          </div>
          <div
            class="checkbox mb-xs last:mb-0 !text-xs xxxl:!text-sm"
          >
            <input
              class="checkbox__input"
              data-testid="checkbox"
              id="3"
              tabindex="0"
              type="checkbox"
              value="Third option"
            />
            <label
              class="checkbox__label"
              for="3"
            >
              Third option
            </label>
          </div>
        </fieldset>
        <div
          class="flex justify-end"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] w-[80px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
            data-testid="button"
            disabled=""
            type="submit"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Save
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`StatementQuestion Multiple choice dropdown displays select without initial answer 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="border border-neutral-300 rounded-xs"
    >
      <div
        class="p-xs pt-x xxxl:p-sm xxxl:pb-x bg-neutral-200 rounded-t-xs text-xs xxxl:text-sm font-medium leading-lg"
      >
        Example question
      </div>
      <div
        class="flex py-sm px-x xxxl:px-sm items-center gap-sm"
      >
        <label
          class="flex flex-col leading-lg min-w-[120px] flex-1 gap-xxxs"
        >
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class="!rounded-sm text-font-secondary !cursor-pointer outline-none !shadow-none !min-h-0 !border-neutral-300 hover:!border-neutral-400 !bg-white select__control css-13cymwt-control"
            >
              <div
                class="flex gap-xxs items-center !flex-nowrap !text-xxs !p-xs !pr-0 !py-xxs h-[30px] select__value-container select__value-container--is-multi css-1fdsijx-ValueContainer"
              >
                <div
                  class="!text-font-secondary !leading-[22px] h-[22px] select__placeholder css-1jqq78o-placeholder"
                  id="react-select-2-placeholder"
                >
                  Select...
                </div>
                <div
                  class="!p-0 !m-0 !leading-[22px] h-[22px] select__input-container css-qbdosj-Input"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-describedby="react-select-2-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class="select__input"
                    id="react-select-2-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="!p-0 !pr-xs select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="!p-0 select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="flex align-center justify-center grow-0 shrink-0 text-neutral-400"
                      data-testid="icon"
                    >
                      <div
                        class="h-sm w-sm"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </label>
        <div
          class="flex justify-end"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] w-[80px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
            data-testid="button"
            disabled=""
            type="submit"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Save
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`StatementQuestion Short text displays text input without initial answer 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="border border-neutral-300 rounded-xs"
    >
      <div
        class="p-xs pt-x xxxl:p-sm xxxl:pb-x bg-neutral-200 rounded-t-xs text-xs xxxl:text-sm font-medium leading-lg"
      >
        Example question
      </div>
      <div
        class="flex py-sm px-x xxxl:px-sm flex-col gap-xs"
      >
        <label
          class="flex flex-col items-start text-font-primary w-full gap-xxxs"
        >
          <div
            class="w-full flex gap-xs items-center justify-between bg-white rounded-sm border border-neutral-300 hover:border-neutral-400 group focus-within:!border-primary-500 p-xs"
          >
            <span
              class="flex gap-xxs grow"
            >
              <input
                class="bg-transparent text-font-primary min-w-0 w-full grow w-full outline-none transition-colors autofill:shadow-[inset_0_0_0px_1000px_rgb(252,252,252)] placeholder:text-font-secondary leading-[14px] text-xxs"
                name="answer"
                placeholder="Type here..."
                type="text"
                value=""
              />
            </span>
          </div>
        </label>
        <div
          class="flex justify-end"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] w-[80px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
            data-testid="button"
            disabled=""
            type="submit"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Save
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`StatementQuestion Single choice displays radio buttons without initial answer 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="border border-neutral-300 rounded-xs"
    >
      <div
        class="p-xs pt-x xxxl:p-sm xxxl:pb-x bg-neutral-200 rounded-t-xs text-xs xxxl:text-sm font-medium leading-lg"
      >
        Example question
      </div>
      <div
        class="flex py-sm px-x xxxl:px-sm flex-col gap-xs"
      >
        <fieldset
          class="p-0"
          role="radiogroup"
        >
          <label
            class="radioLabel mb-xs last:mb-0 text-xs xxxl:text-sm"
          >
            <input
              class="radioInput"
              name="answer"
              type="radio"
              value="First option"
            />
            <div
              class="radioSpan"
            >
              <span
                class="labelWrapper"
              >
                First option
              </span>
            </div>
          </label>
          <label
            class="radioLabel mb-xs last:mb-0 text-xs xxxl:text-sm"
          >
            <input
              class="radioInput"
              name="answer"
              type="radio"
              value="Second option"
            />
            <div
              class="radioSpan"
            >
              <span
                class="labelWrapper"
              >
                Second option
              </span>
            </div>
          </label>
          <label
            class="radioLabel mb-xs last:mb-0 text-xs xxxl:text-sm"
          >
            <input
              class="radioInput"
              name="answer"
              type="radio"
              value="Third option"
            />
            <div
              class="radioSpan"
            >
              <span
                class="labelWrapper"
              >
                Third option
              </span>
            </div>
          </label>
        </fieldset>
        <div
          class="flex justify-end"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] w-[80px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
            data-testid="button"
            disabled=""
            type="submit"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Save
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
